src/main/java/com/cloudbees/jenkins/support/impl/AboutJenkins.java

```diff
@@ -439,7 +439,7 @@ private String getInfo(ContentFilter filter) {
                         .append(" arg[")
                         .append(count++)
                         .append("]: `")
-                        .append(Markdown.escapeBacktick(ContentFilter.filter(filter, arg)))
+                        .append(Markdown.escapeBacktick(filter != null ? ContentFilter.filter(filter, arg) : arg))
```

Comment:
Turning on fine logging revealed

```
FINE	c.c.j.support.AsyncResultCache#get: Could not retrieve Java info from agent0
Also:   hudson.remoting.Channel$CallSiteStackTrace: Remote call to agent0
		at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1784)
		at hudson.remoting.UserRequest$ExceptionResponse.retrieve(UserRequest.java:356)
		at hudson.remoting.Channel$2.adapt(Channel.java:1034)
		at hudson.remoting.Channel$2.adapt(Channel.java:1030)
		at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
		at com.cloudbees.jenkins.support.AsyncResultCache.get(AsyncResultCache.java:69)
		at com.cloudbees.jenkins.support.impl.AboutJenkins$NodesContent.printTo(AboutJenkins.java:808)
		at com.cloudbees.jenkins.support.api.PrefilteredPrintedContent.writeTo(PrefilteredPrintedContent.java:63)
		at com.cloudbees.jenkins.support.SupportPlugin.writeBundle(SupportPlugin.java:420)
		at com.cloudbees.jenkins.support.SupportPlugin.writeBundle(SupportPlugin.java:355)
		at com.cloudbees.jenkins.support.CheckFilterTest.assertComponent(CheckFilterTest.java:158)
		at com.cloudbees.jenkins.support.CheckFilterTest.checkFilterTest(CheckFilterTest.java:92)
		at …
java.lang.NullPointerException
	at com.cloudbees.jenkins.support.filter.ContentFilter.filter(ContentFilter.java:109)
	at com.cloudbees.jenkins.support.impl.AboutJenkins$GetJavaInfo.getInfo(AboutJenkins.java:442)
	at com.cloudbees.jenkins.support.impl.AboutJenkins$GetJavaInfo.call(AboutJenkins.java:227)
	at com.cloudbees.jenkins.support.impl.AboutJenkins$GetJavaInfo.call(AboutJenkins.java:214)
	at hudson.remoting.UserRequest.perform(UserRequest.java:211)
	at …
Caused: java.util.concurrent.ExecutionException
	at hudson.remoting.Channel$2.adapt(Channel.java:1036)
	at hudson.remoting.Channel$2.adapt(Channel.java:1030)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at com.cloudbees.jenkins.support.AsyncResultCache.get(AsyncResultCache.java:69)
	at com.cloudbees.jenkins.support.impl.AboutJenkins$NodesContent.printTo(AboutJenkins.java:808)
	at com.cloudbees.jenkins.support.api.PrefilteredPrintedContent.writeTo(PrefilteredPrintedContent.java:63)
	at com.cloudbees.jenkins.support.SupportPlugin.writeBundle(SupportPlugin.java:420)
	at com.cloudbees.jenkins.support.SupportPlugin.writeBundle(SupportPlugin.java:355)
	at com.cloudbees.jenkins.support.CheckFilterTest.assertComponent(CheckFilterTest.java:158)
	at com.cloudbees.jenkins.support.CheckFilterTest.checkFilterTest(CheckFilterTest.java:92)
	at …
```

Reply:
so this somehow is a production bug.  the filter should be filtering (anonymizing) sensitive info - and this if a JNLP agent may contain the Jenkins URL (which should be anonymized) but actually never can be...?

Reply:
I think https://github.com/jenkinsci/support-core-plugin/blob/14c7b90837029491ab279269a3bfe56cfd57e0f5/src/main/java/com/cloudbees/jenkins/support/impl/AboutJenkins.java#L816-L820 (#176) does that? It looks like #502 introduced the NPE.

Reply:
I think this might have always been broken... See https://github.com/jenkinsci/support-core-plugin/pull/473#discussion_r1231700229.

Reply:
so for PII / security reasons we should (if the filter is null) not include the args at all (at least as a temporary fix to allow the deflaking of this until the code can be propertly fixed in a different ticket?  (that would IIUC not loose any info from agents as that info would not be available today as the code would have have blown up?)

Reply:
👍  I mistakenly thought this was a partial cause of the flake (if an agent was connected at the time the bundle ran)

Reply:
No, probably unrelated (the flake was about the _master_ node, weirdly enough), just something I saw when I turned on `FINE` logging.